ActionSheet.android.js

```diff
@@ -87,19 +96,24 @@ class ActionGroup extends React.Component {
     );
 
     for (let i = startIndex; i < startIndex + length; i++) {
-      let color = '#212121';
-      if (i === destructiveButtonIndex) {
-        color = '#d32f2f';
-      }
-
+      const defaultColor = tintColor
+        ? tintColor
+        : (textStyle || {}).color || BLACK_87PC_TRANSPARENT;
+      const color = i === destructiveButtonIndex ? '#d32f2f' : defaultColor;
+      const iconSource = icons != null && icons[i];
       let iconElement = undefined;
 
-      if (icons && icons[i]) {
-        const iconStyle = [styles.icon]
-        if (textStyle.color !== undefined && textStyle.color !== null) {
-          iconStyle.push({ tintColor: textStyle.color })
+      if (iconSource) {
+        if (typeof iconSource === 'number') {
+          const iconStyle = [styles.icon, { tintColor: tintIcons ? color : null }];
+          iconElement = <Image
+            fadeDuration={0}
+            source={iconSource}
+            resizeMode="contain"
+            style={iconStyle}/>;
+        } else {
+          iconElement = <View style={styles.icon}>{iconSource}</View>;
```

Comment:
what exactly is `iconSource` in this case? an React element?

Reply:
Either a required image source, or a React element, correct.